examples/include/example_helper.hpp

```diff
@@ -67,4 +75,57 @@ void rand_matrix(vec &M, oneapi::mkl::transpose trans, int m, int n, int ld) {
     }
 }
 
+template <typename fp, typename intType>
+void generate_sparse_matrix(const intType nx, intType *ia, intType *ja, fp *a,
+                            const intType index = 0) {
```

Comment:
I'd suggest:
```suggestion
template <typename fp, typename intType>
intType generate_sparse_matrix(const intType nx, intType *ia, intType *ja, fp *a,
                            const intType index = 0) {
```
followed by `return ia[nx];` at the end of the function. This returns the actual number of non-zeros generated in the matrix. The way we use this function is that we over-allocate memory for `ja`, `a`, fill most of it up, and rely on the exact non-zero count to be stored in `ia[nx]`.

Reply:
I have re-used the same example from mkl > sycl, just like the other domains do. I don't see `generate_sparse_matrix` being used that way even in the most recent version. I'd rather keep them as close as possible.